test/regress/cover24.sh

```diff
@@ -0,0 +1,14 @@
+set -xe
```

Comment:
If this test doesn't actually verify anything then there's no point having it run in the regression tests. It's fine to just verify once manually that the bug is fixed.

Reply:
Actually it does. With the first implementation, I had difference between numbers in NEORV32 hiearchy/file report.

The issue was, that if a cover scope had invalid loc, it did not process nested locs. Then, cover items
for else statement in `when / else` construct were not re-shuffled to per-file lists correctly. This was since
`T_COND_STMT` parent of `T_SIGNAL_ASSIGN` had no loc (which is correct since there is no condition).

So, the test-case above is MWE for this case. Before the fix, it showed one stmt cover bin less.
Its just the implementation that mixed together the original implementation and bugfix.

Reply:
plus, if I leave the test out, the coverage will drop since there would be no test in regression that does
the `per-file` report.